File: leetcode/solutions/0064.MinimumPathSum/minPathSum_test.go
package solution

import (
	"testing"
)

func Test_0001_TwoSum(t *testing.T) {
	t.Parallel()

	type in struct {
		arg1 []int
		arg2 int
	}
	tests := []struct {
		in  [][]int
		out int
	}{
		{
			[][]int{
				{1, 3, 1},
				{1, 5, 1},
				{4, 2, 1},
			},
			7,
		},
		{
			[][]int{
				{1},
			},
			1,
		},
		{
			[][]int{
				{0},
			},
			0,
		},
		{
			[][]int{
				{1, 1},
				{2, 1},
			},
			3,
		},
	}

	for _, tt := range tests {
		tt := tt

		t.Run("Test", func(t *testing.T) {
			t.Parallel()

			got := minPathSum(tt.in)

			if got != tt.out {
				t.Errorf("got %#v want %#v", got, tt.out)
			}
		})
	}
}

File: leetcode/solutions/0064.MinimumPathSum/minPathSum_test.ts
import { assertEquals } from "https://deno.land/std/testing/asserts.ts";
import minPathSum from "./minPathSum.ts";

Deno.test("0064. Minimum Path Sum", () => {
  type test = [number[][], number];
  const tests: test[] = [
    [
      [
        [1, 3, 1],
        [1, 5, 1],
        [4, 2, 1],
      ],
      7,
    ],
    [[[1]], 1],
    [[[0]], 0],
    [
      [
        [1, 1],
        [2, 1],
      ],
      3,
    ],
    [
      [
        [1, 2, 5],
        [3, 2, 1],
      ],
      6,
    ],
  ];

  for (const [arg, want] of tests) {
    assertEquals(minPathSum(arg), want);
  }
});
